ocs_ci/utility/aws.py

```diff
@@ -262,50 +268,79 @@ def detach_and_delete_volume(self, volume, timeout=120):
             delete_response
         )
 
-    def stop_ec2_instance(self, instance_id, wait=False):
+    def stop_ec2_instances(self, instances, wait=False, force=True):
         """
         Stopping an instance
 
         Args:
-            instance_id (str): ID of the instance to stop
+            instances (dict): A dictionary of instance IDs and names to stop
             wait (bool): True in case wait for status is needed,
                 False otherwise
+            force (bool): True for force instance stop, False otherwise
 
-        Returns:
-            bool: True in case operation succeeded, False otherwise
         """
-        res = self.ec2_client.stop_instances(
-            InstanceIds=[instance_id], Force=True
-        )
+        instance_ids, instance_names = zip(*instances.items())
+        logger.info(f"Stopping instances {instance_names} with Force={force}")
+        ret = self.ec2_client.stop_instances(InstanceIds=instance_ids, Force=force)
+        stopping_instances = ret.get('StoppingInstances')
+        for instance in stopping_instances:
+            assert instance.get('CurrentState').get('Code') in [
+                constants.INSTANCE_STOPPED, constants.INSTANCE_STOPPING,
+                constants.INSTANCE_SHUTTING_DOWN
+            ], (
+                f"Instance {instance.get('InstanceId')} status "
+                f"is {instance.get('CurrentState').get('Code')}"
+            )
         if wait:
-            for sample in utils.TimeoutSampler(
-                TIMEOUT, SLEEP, self.get_instances_status_by_id, instance_id
-            ):
-                if sample == STOPPED:
-                    return True
-            return False
-        state = res.get('StoppingInstances')[0].get('CurrentState').get('Code')
-        return state == STOPPING
-
-    def start_ec2_instance(self, instance_id, wait=False):
+            for instance_id, instance_name in instances.items():
+                logger.info(
+                    f"Waiting for instance {instance_name} to reach status stopped"
+                )
+                instance = self.get_ec2_instance(instance_id)
+                instance.wait_until_stopped()
```

Comment:
Just to make sure - if we'll receive a timeout, the test *should* fail and go to teardown, right? There's no need for try-except

Reply:
in case of more timeout (10 minutes), we'll go directly to teardown